app/src/main/java/com/arctouch/codechallenge/view/movie/MovieFragment.kt

```diff
@@ -0,0 +1,96 @@
+package com.arctouch.codechallenge.view.movie
+
+import android.annotation.SuppressLint
+import android.os.Bundle
+import android.transition.TransitionInflater
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.view.ViewCompat
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProviders
+import androidx.lifecycle.lifecycleScope
+import androidx.navigation.fragment.navArgs
+import com.arctouch.codechallenge.R
+import com.arctouch.codechallenge.databinding.FragmentMovieBinding
+import com.arctouch.codechallenge.model.data.Genre
+import com.arctouch.codechallenge.view.MainActivity
+import com.arctouch.codechallenge.view.databinding.buildBackdropUrl
+import com.arctouch.codechallenge.view.databinding.buildPosterUrl
+import org.koin.android.ext.android.inject
+
+class MovieFragment : Fragment() {
+
+    private val factory: MovieViewModel.Factory by inject()
+    private val viewModel: MovieViewModel by lazy {
+        ViewModelProviders.of(this, factory)
+                .get(MovieViewModel::class.java)
+    }
+
+    private val args: MovieFragmentArgs by navArgs()
+
+    init {
+        lifecycleScope.launchWhenCreated {
+            viewModel.getMovieById()
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        sharedElementEnterTransition = TransitionInflater.from(requireContext())
+                .inflateTransition(R.transition.default_transition)
+    }
+
+    @SuppressLint("SetTextI18n")
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        val binding = FragmentMovieBinding.inflate(inflater, container, false)
+        val movieId = args.movieId
+        val movieName = args.movieName
+
+        (activity as MainActivity).supportActionBar?.apply {
+            setDisplayHomeAsUpEnabled(true)
+            title = movieName
+
+        }
+        factory.setMovieID(movieId)
+
+        binding.lifecycleOwner = this
+        binding.viewModel = viewModel
+
+        viewModel.movie.observe(this, Observer { movie ->
+            createTags(binding, movie.genres)
+            binding.description.text = movie.title
+            binding.description.text = if (!movie.overview.isNullOrEmpty()) movie.overview else "Descrição não disponível"
```

Comment:
How could we make sure we get the fallback text for `movie.overview` (which is hardcoded here) in the correct language?

Reply:
We can create a file called strings.xml passing the language for the country (which at first time we can use the default from device) and create an option to user set a location to retrieve the information from different region

Reply:
If using the default `strings.xml` and language-specific resources with translations, Android will handle that for you automatically. More on that here: https://developer.android.com/guide/topics/resources/localization#creating-alternatives and here: https://developer.android.com/training/basics/supporting-devices/languages#MirroringAddResources

Reply:
You are correctly, that's what I mean when I said create a strings.xml, the second idea I followed what I have in a project which allows user to use different language when needed.